### PR TITLE
feat: use ADD for $inc

### PR DESCRIPTION
**Pull request by onhate**
[Expressions.UpdateExpressions.ADD](https://docs.aws.amazon.com/amazondynamodb/latest/developerguide/Expressions.UpdateExpressions.html#Expressions.UpdateExpressions.ADD)

### CODE REVIEW

**Comment by onhate** [comment]
codecov decreased because lines of code reduced too.

**Comment by onhate** [comment]
@SamVerschueren what do you think about this feature?

**Comment by onhate** [comment]
@SamVerschueren ?

**Comment by SamVerschueren** [comment]
Hi @onhate. Sorry about the delay.

I noticed the following comment in the docs though.

image

Not sure why, it's not really explained. The change for using `ADD` should behave exactly the same as what we had with `SET`, right? The only benefit is that we don't have to generate the `_v_empty_value`?

**Comment by onhate** [comment]
🤔  that's weird, I think they meant that because ADD can have different behaviors if using it with sets or numbers which could cause confusion, like trying to create a set with a number and instead it creates a number field, but on our case we have clear operations for sets($addToSet, $removeFromSet) and numbers ($inc) that will enforce the creation of a  set or a numeric value.

Correct, the main benefit is to remove the extra overhead of checking if exists or not and the _empty_value.

File: src/test/methods/upsert.ts [hunk 1]
```diff
@@ -121,13 +121,12 @@ test.serial('increment and set with upsert', async t => {
 		Key: {
 			id: '5'
 		},
-		UpdateExpression: 'SET #k_foo=:v_foo, #k_salary=if_not_exists(#k_salary, :_v_empty_value)+:v_salary',
+		UpdateExpression: 'SET #k_foo=:v_foo ADD #k_salary :v_salary',
```
Comment (SamVerschueren) [inline src/test/methods/upsert.ts]: Can you add an extra property to the `$inc` operator? Just to see how it would work with multiple values.

**Review by SamVerschueren — CHANGES_REQUESTED** [review]

File: src/test/methods/upsert.ts (hunk 1 above)
Comment (onhate) [reply src/test/methods/upsert.ts]: done

**Review by SamVerschueren — APPROVED** [review]

**Pull request merged**